src/main/java/com/facebook/openwifirrm/modules/ConfigManager.java

```diff
@@ -364,12 +392,44 @@ public boolean removeConfigListener(String id) {
 		return (configListeners.remove(id) != null);
 	}
 
-	/** Interrupt the main thread, possibly triggering an update immediately. */
+	/**
+	 * Track the zone to be updated in the future.
+	 *
+	 * @param zone zone (i.e., venue), or null, which indicates all zones
+	 */
+	public void queueForUpdate(String zone) {
+		if (zone != null) {
+			temporaryZonesToUpdateBuffer.add(zone);
+		} else {
+			/*
+			 * Here, addAll is not atomic, so read operations during the addAll
+			 * may read none, some, or all of the items passed in to addAll.
+			 * But, it is ok if different zones are updated at different times.
+			 */
+			temporaryZonesToUpdateBuffer.addAll(deviceDataManager.getZones());
+		}
+	}
+
+	/**
+	 * Interrupt the main thread, possibly triggering an update immediately.
+	 */
 	public void wakeUp() {
-		eventFlag.set(true);
+		zonesToUpdate.addAll(temporaryZonesToUpdateBuffer);
+		temporaryZonesToUpdateBuffer.clear();
```

Comment:
as mentioned above, totally don't see the purpose of the second queue or this code here